cpp/include/cudf/strings/strings_column_view.hpp

```diff
@@ -26,23 +26,22 @@ namespace cudf {
  * @brief Given a column-view of strings type, an instance of this class
  * provides a wrapper on this compound column for strings operations.
  *---------------------------------------------------------------------------**/
-class strings_column_view : private column_view
+class strings_column_view : public column_view
```

Comment:
By using `public` inheritance, you're pulling along every member function in `column_view`. Even functions that don't make sense for a strings column, e.g., `head()`, `data()`, etc.

The reason I originally suggested private inheritance was so that one could selectively enable member functions from `column_view` via the `using` mechanism, e.g.,
```
class strings_column_view : private column_view{
   // because of private inheritance, none of the members of `column_view` are available by default

  // You can selectively enable members of `column_view` like this
  using column_view::size; // 
}
```